packages/neo-one-client/src/user/keystore/LedgerKeyStore.ts

```diff
@@ -0,0 +1,232 @@
+        );
+
+        return {
+          verification: witness.verification.toString('hex'),
+          invocation: witness.invocation.toString('hex'),
+        };
+      },
+      'neo_ledger_sign',
+      monitor,
+    );
+  }
+
+  public async selectAccount(id?: UserAccountID, _monitor?: Monitor): Promise<void> {
+    if (id === undefined) {
+      this.currentAccountInternal$.next(undefined);
+    } else {
+      const { account } = this.getLedger(id);
+      this.currentAccountInternal$.next(account);
+    }
+  }
+
+  public getLedger({ address, network }: UserAccountID): Ledger {
+    const ledgers = this.ledgers[network];
+    if (ledgers === undefined) {
+      throw new UnknownAccountError(address);
+    }
+
+    const ledger = ledgers[address];
+    if (ledger === undefined) {
+      throw new UnknownAccountError(address);
+    }
+
+    return ledger;
+  }
+
+  public getLedgers$({ address, network }: UserAccountID): Observable<Ledger | undefined> {
+    return this.ledgersInternal$.pipe(
+      map((ledgers) => {
+        const networkLedgers = ledgers[network];
+        if (networkLedgers === undefined) {
+          return undefined;
+        }
+
+        return networkLedgers[address];
+      }),
+    );
+  }
+  // we just init the first account on the ledger for now, ideally we have some kind of scanning method to extract actual accounts.
+  public async init(): Promise<LedgerHandler> {
+    const handler = await LedgerHandler.init();
+    const accounts = await this.scanLedgerAccounts(handler);
+
+    this.initWithLedgers(accounts);
+
+    return handler;
+  }
+
+  public async close(): Promise<void> {
+    const handler = await this.initPromise;
+    await handler.close();
+  }
+
+  private initWithLedgers(accountsToLedgers: ReadonlyArray<Ledger>): void {
+    const ledgers = accountsToLedgers.reduce<Ledgers>(
+      (acc, ledger) => ({
+        ...acc,
+        [ledger.account.id.network]: {
+          ...(acc[ledger.account.id.network] === undefined ? {} : acc[ledger.account.id.network]),
+          [ledger.account.id.address]: ledger,
+        },
+      }),
+      {},
+    );
+
+    this.ledgersInternal$.next(ledgers);
+    this.newCurrentAccount();
+  }
+
+  private newCurrentAccount(): void {
+    const allAccounts = flattenLedgers(this.ledgers).map(({ account: value }) => value);
+
+    const account = allAccounts[allAccounts.length - 2];
+    this.currentAccountInternal$.next(account);
+  }
+
+  private async scanLedgerAccounts(handler: LedgerHandler, maxOffset = 20): Promise<ReadonlyArray<Ledger>> {
+    const scanAccountsInternal = async (start: number, currentOffset = 0): Promise<ReadonlyArray<Ledger>> => {
+      if (currentOffset <= maxOffset) {
+        const publicKey = await handler.getPublicKey(start + currentOffset);
+        const ledgerAccount = this.publicKeyToLedgerAccount(start + currentOffset, publicKey);
+        const account = await this.provider.getAccount(ledgerAccount.account.id.address);
```

Comment:
You actually do want to batch these into groups of 20 - otherwise we're sending 20 individual `getAccount` requests and it's going to be quite slow waiting for 20 round trips.